service/machine.py

```diff
@@ -195,12 +195,26 @@ def share_with_admins(private_userlist, provider_uuid):
 
 
 def upload_privacy_data(machine_request, new_machine):
+    """
+    ASSERT: The image in 'new_machine' SHOULD BE private
+    (Based on values in machine_request)
+    """
     prov = new_machine.provider
     accounts = get_account_driver(prov)
     if not accounts:
         print "Aborting import: Could not retrieve Account Driver "\
             "for Provider %s" % prov
         return
+    img = accounts.get_image(new_machine.identifier)
+    if hasattr(img, 'visibility'):  # Treated as an obj.
+        is_public = img.visibility == 'public'
+    elif hasattr(img, 'items'):  # Treated as a dict.
+        is_public = img.get('visibility','N/A') == 'public'
+
+    if is_public:
+        print "Marking image %s private" % img.id
+        accounts.image_manager.update_image(img, visibility='private')
+
```

Comment:
I'm confused about how this is _private_ when the conditional is testing for it being public. I'm sure there is some contextual knowledge I lack.